File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        def productOfDigits(num):
            product = 1
            while num:
                product *= num % 10
                num //= 10
            return product

        while productOfDigits(n) % t != 0:
            n += 1
        return n

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.java
class Solution {
    public int smallestNumber(int n, int t) {
        while(productOfDigits(n) % t != 0){
            n++;
        }
        return n;
    }

    private int productOfDigits(int num){
        int product = 1;
        while(num != 0){
            product *= num % 10;
            num /= 10;
        }
        return product;
    }
}